lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -0,0 +1,154 @@
+import { LabeledCheckbox } from '@hypothesis/frontend-shared';
+import { Fragment, createElement } from 'preact';
+import { useCallback, useContext, useEffect, useState } from 'preact/hooks';
+
+import { Config } from '../config';
+import { apiCall } from '../utils/api';
+import { useUniqueId } from '../utils/hooks';
+
+import AuthButton from './AuthButton';
+
+/**
+ * @typedef {import('../api-types').GroupSet} GroupSet
+ */
+
+/**
+ * Configuration relating to how students are divided into groups for an
+ * assignment.
+ *
+ * @typedef GroupConfig
+ * @prop {boolean} useGroups - Whether students are divided into groups
+ * @prop {string|null} groupSet - The ID of the grouping to use. This should
+ *   be the `id` of a `GroupSet` returned by the LMS backend.
+ */
+
+/**
+ * @typedef GroupSetSelectorProps
+ * @prop {GroupConfig} groupConfig
+ * @prop {(g: GroupConfig) => void} onChangeGroupConfig
+ */
+
+/**
+ * Component that allows instructors to configure how students are divided into
+ * groups for an assignment.
+ *
+ * @param {GroupSetSelectorProps} props
+ */
+export default function GroupConfigSelector({
+  groupConfig,
+  onChangeGroupConfig,
+}) {
+  const [groupSets, setGroupSets] = useState(
+    /** @type {GroupSet[]|null} */ (null)
+  );
+
+  const [fetchError, setFetchError] = useState(
+    /** @type {Error|null} */ (null)
+  );
+
+  const {
+    api: { authToken },
+    filePicker: {
+      canvas: { listGroupSets: listGroupSetsAPI },
+    },
+  } = useContext(Config);
+
+  const fetchGroupSets = useCallback(async () => {
+    setFetchError(null);
+
+    try {
+      const groupSets = /** @type {GroupSet[]} */ (
+        await apiCall({
+          authToken,
+          path: listGroupSetsAPI.path,
+        })
+      );
+      setGroupSets(groupSets);
+    } catch (err) {
+      setFetchError(err);
+    }
+  }, [authToken, listGroupSetsAPI]);
+
+  const useGroups = groupConfig.useGroups;
+  const haveFetchedGroups = groupSets !== null;
+
+  useEffect(() => {
+    if (useGroups && !haveFetchedGroups) {
+      fetchGroupSets();
+    }
+  }, [fetchGroupSets, useGroups, haveFetchedGroups]);
+
+  const checkboxID = useUniqueId('GroupSetSelector__enabled');
+  const selectID = useUniqueId('GroupSetSelector__select');
+
+  const groupSet = groupConfig.useGroups ? groupConfig.groupSet : null;
+  const fetchingGroupSets = !groupSets && !fetchError && useGroups;
+
+  return (
+    <Fragment>
+      <div>
+        <LabeledCheckbox
+          checked={useGroups}
+          id={checkboxID}
+          name="groupsEnabled"
```

Comment:
This name confuses me! Is it related to the `groupsEnabled` Canvas-specific configuration? If so, can you help me understand? If not, can we change it :) ?

Reply:
The `name` prop is required by `LabeledCheckbox` for use with the `<input>` but doesn't actually matter as the field is not part of the submitted form. I'll add a note to clarify.